### PR TITLE
feat: bootstrapping and webpack docs

### PR DESCRIPTION
**Pull request by markuczy**

### CODE REVIEW

File: docs/modules/general/pages/bootstrapping.adoc [hunk 1]
```diff
@@ -0,0 +1,567 @@
+= OneCX Microfrontend Content Bootstrapping
+
+== Overview
+// TODO: Add link to MFE loading
+This document aims to explain the details related to Microfrontend's content bootstrapping. Depending on the content's type and expose method there are different requirements needed to be fullfield to ensure the content is loaded correctly by OneCX platform elements. This topic is highly related to Microfrontend content loading and xref:webpack.adoc[Microfrontend configuration with Webpack]. It is recommended to be familiar with those before reading this document.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:5]: Depending on the content's type and expose method **,** there are different requirements needed to be ~~fullfield~~ **fulfilled** to...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+
+== Microfrontend's Content Bootstrapping Explained
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:7]: OneCX Microfrontend ~~'s~~ Content Bootstrapping --> Microfrontend's Content or Microfrontend Content? Probably means the same at the end, the latter may be easier to read

File: docs/modules/general/pages/bootstrapping.adoc [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+// TODO: Add link to MFE loading
+Microfrontend's content loading is an action triggered by the route change (Shell Application's responsibility of loading modules) or by the Slot Component display (Slot Component's responsibility of loading Remote Components). During this action Microfrontend's content, exposed via remote entry file, is loaded and bootstrapped using the provided file in xref:webpack.adoc[Webpack configuration].
+
+Depending on the content type and chosen expose method, different adjustments have to be made to ensure correct content bootstrapping. 
+
+=== Content Exposed Using Angular Method
+IMPORTANT: **It is NOT recommended** to use the Angular expose method. Doing so restrains the platform from using many frameworks with different versions, which means that all Microfrontends would need to use the same technology and version.
+
+When exposing modules and Remote Components using the Angular method, there are two requirements.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:16]: there are two requirements **:**

File: docs/modules/general/pages/bootstrapping.adoc [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+  enableProdMode();
+}
+
+platformBrowserDynamic().bootstrapModule(AppModule)
+  .catch(err => console.error(err));
+....
+
+=== Content Exposed Using Webcomponent Method
+// TODO: Correct link to exact resource for @onecx/angular-webcomponents lib
+xref:implementation-details/mfe-content-loading/webcomponents.adoc[Loading mechanism for modules and Remote Components exposed using Webcomponent method] requires that content is adapted to using Web Components Custom Elements. For both content types, bootstrapping logic needs to register Custom Element in the Custom Elements Registry.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:44]: ... logic needs to register **a** Custom Element in the ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 5]
```diff
@@ -0,0 +1,567 @@
...
+
+// TODO: Correct link to exact resource for Angular issues
+NOTE: For Angular based content, apart from Custom Element registration it is also required to cover a few Angular related issues. For more information on those issues, please refer xref:implementation-details/mfe-content-loading/webcomponents.adoc[here]
+
+// TODO: Correct link to exact resource for @onecx/angular-webcomponents lib
+It is recommended to use xref:implementation-details/mfe-content-loading/webcomponents.adoc[@onecx/angular-webcomponents] library to bootstrap Microfrontend's content using its exposed methods. This library will be used in the examples throughout this document.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:50]: It is recommended to use **the** ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+
+==== Module Bootstrapping and Configuration
+The following requirements need to be fullfield when exposing modules using Webcomponent method:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:53]: The following requirements need to be ~~fullfield~~ **fulfilled** when exposing modules using **the** Webcomponent method:

File: docs/modules/general/pages/bootstrapping.adoc [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+            deps: [Router, AppStateService]
+        },
+        provideHttpClient(withInterceptorsFromDi())
+    ]
+})
+export class ExampleModule implements DoBootstrap {
+    constructor(private readonly injector: Injector) {}
+
+    ngDoBootstrap(): void {
+        createAppEntrypoint(AppEntrypointComponent, 'example-component', this.injector)
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:189]: createAppEntrypoint(AppEntrypointComponent, 'example-**web**(?)component', this.injector)

File: docs/modules/general/pages/bootstrapping.adoc [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+    }
+}
+....
+
+The **module.ts** file is prepares the module for integration with OneCX platform.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:194]: The **remote.(?)module.ts**  file ~~is~~ prepares the module for integration with **the** OneCX platform.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 9]
```diff
@@ -0,0 +1,567 @@
...
+
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:195]: maybe create a section for the following. **Remote Module example deep dive** or something similar

File: docs/modules/general/pages/bootstrapping.adoc [hunk 10]
```diff
@@ -0,0 +1,567 @@
...
+This example showcases the recommended approach of defining modules (Angular based) using Webcomponent method. Here is a list of important features of this example:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:196]: ... using **the** Webcomponent method.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 11]
```diff
@@ -0,0 +1,567 @@
...
+* CommonModule, BrowserModule and BrowserAnimationsModule Angular modules used for adding functionality to the module.
+* PortalCoreModule is defined to allow usage of OneCX components and services.
+* TranslateModule is defined to allow translations using translation keys within the module.
+// TODO: Add link to auth docs
+* AngularAuthModule is defined to use OneCX authorization mechanisms.
+* RouterModule is defined for routing to feature modules within the exposed module.
+
+Entrypoint component::
+// TODO: Add link to createAppEntrypoint
+AppEntrypoint is a standard Angular component, which has a `<router-outlet>` element in its template. The `createAppEntrypoint` registers AppEntrypointComponent in the Custom Elements Registry, so anytime '<example-component>' is rendered, AppEntrypointComponent should be instantiated.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:208]: ... <example-**web**(?)component> ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 12]
```diff
@@ -0,0 +1,567 @@
...
++
+The third parameter, being the module's Injector, is very important. This injector will be used by the instances of AppEntrypointComponent rendered using Web Components technology, meaning that each instance will have everything related to the module already set up. That also means the `<router-outlet>` will be using routes defined for the module.
++
+// TODO: Add link to createAppEntrypoint
+// TODO: Link to events topic
+[[module-rotuer-connection]]The `createAppEntrypoint` method is also responsible for connecting the module's router to Shell's router. Every time the URL of the browser changes, Shell is going to publish a new message, via EventsTopic, with information about the new URL. The `createAppEntrypoint` method subscribes to the EventsTopic and updates the router state accordingly to the received information.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:214]: ... connecting the module's router to **the** Shell's router. Every time the URL of the browser changes, **the** Shell is going to publish a new message ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 13]
```diff
@@ -0,0 +1,567 @@
...
++
+For technologies other than Angular, it is recommended to:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:216]: maybe use here a NOTE: ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 14]
```diff
@@ -0,0 +1,567 @@
...
+* User enters 'shell_url/workspace_name/example_base_path' URL - FeatureModule is used.
+* User enters 'shell_url/workspace_name/example_base_path/tracking' URL - TrackingModule is used.
+--
++
+With the following URL parts meaning:
++
+--
+* shell_url - the Shell Application deployment URL, e.g. `localhost:4200/shell.`
+* workspace_name - name of the accessed Workspace, e.g. `admin`.
+* example_base_path - base path of the example Microfrontend (configured via OneCX Core Applications), e.g. `example`.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:239]: `shell_url`
`workspace_name`
`example_base_path`
or use any other kind of formatting

File: docs/modules/general/pages/bootstrapping.adoc [hunk 15]
```diff
@@ -0,0 +1,567 @@
...
+** displays nothing (for a period of time when the user routes between Microfrontends).
+--
++
+// TODO: link to statsWith
+The `startsWith` method uses Microfrontend information, saved in the route's data, to remove already used parts from consideration when matching routes within the module.
++
+// TODO: link to lib
+To create your own matchers, please consider using `@onecx/angular-webcomponents` library.
++
+For technologies other than Angular, it is recommended to:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:266]: maybe use here a NOTE: ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 16]
```diff
@@ -0,0 +1,567 @@
...
++
+--
+* Use Microfrontend information from CurrentMfeTopic to only use relevant parts of the URL for routing and redirect correctly.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:269]: ... for routing and redirect**ing** correctly.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+* Ensure routing away from the module is not causing side-effects.
+--
+
+Configuration::
+All services utilizing HttpClient used within the Microfrontend's module need to know how to make requests to external resources. Depending on the configuration of the Workspace, they need to take that context into consideration when creating a URL for those resources.
++
+A service might want to call `deployment_url/bff/search` by default. With this call being made, MFE App will need to access the BFF. When the Application's path within the Workspace is `mfe/example` the call has to be made to `deployment_url/mfe/example/bff/search`.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:276]: With this call being made, **the** MFE App will need to access the BFF.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 18]
```diff
@@ -0,0 +1,567 @@
...
++
+// TODO: Link to Topic
+The `apiConfigProvider` presented in the example is utilizing `PortalApiConfiguration` class as `Configuration` for the services. It is listening to the CurrentMfeTopic changes and overwriting the basePath accordingly to the received message and all services use that information to construct a valid URL.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:279]: The `apiConfigProvider` presented in the example is utilizing **the** `PortalApiConfiguration` class as `Configuration` for the services.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 19]
```diff
@@ -0,0 +1,567 @@
...
++
+For technologies other than Angular, it is recommended to:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:281]: maybe use here a NOTE: ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 20]
```diff
@@ -0,0 +1,567 @@
...
+. Webpack configuration needs to expose the *main.ts* file of the Remote Component.
+. The **main.ts** file of the Remote Component needs to bootstrap the component asynchronously.
+// TODO: Correct link to exact resource for Angular issues
+. Component bootstrap
+.. has to cover Angular issues (for Angular based Remote Components only)
+.. needs to register the component as a Custom Element in the Custom Elements Registry.
+.. needs to adapt routing so Shell registers routing happening within the Remote Component.
+. Component class needs to use the Remote Component initialization mechanism.
+
+Below is a Remote Component setup example (Angular based) for using the Webcomponent expose method is presented. Each file's content is going to be explained in detail with important context information for technologies other than presented.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:299]: Below is a Remote Component setup example (Angular based) for using the Webcomponent expose method ~~is presented~~.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 21]
```diff
@@ -0,0 +1,567 @@
...
+      deps: [UserService],
+      multi: true,
+    },
+  ]
+)
+....
+
+// TODO: Correct link to exact resource for Angular issues
+// TODO: Link to the @onecx/angular-webcomponents
+The **example.component.bootstrap.ts** file is bootstrapping the Remote Component using @onecx/angular-webcomponents library to ensure that Angular ngZone and platform are shared (Angular requirement only). It also connects Shell router with the Remote Component's router (if such exists). The last argument is an array of providers required for the component to work properly. Detailed description of this file can be found in the <<summary>>.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:394]: It also connects **the** Shell router with the Remote Component's router (if such exists)...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 22]
```diff
@@ -0,0 +1,567 @@
...
+    this.baseUrl.next(config.baseUrl);
+    this.permissions = config.permissions;
+    this.exampleService.configuration = new Configuration({
+      basePath: Location.joinWithSlash(config.baseUrl, environment.apiPrefix)
+    })
+  }
+}
+....
+
+The **example.component.ts** file prepares the Remote Component for integration with OneCX platform.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:467]: ... for integration with **the** OneCX platform.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 23]
```diff
@@ -0,0 +1,567 @@
...
+
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:468]: maybe create a section **Summary** like you already mentioned it here

File: docs/modules/general/pages/bootstrapping.adoc [hunk 24]
```diff
@@ -0,0 +1,567 @@
...
+* providers from BrowserAnimationsModule.
+// TODO: Link to userService
+* APP_INITIALIZER using userProfileInitializer factory function - in ExampleComponent's constructor `this.userService.lang$.getValue()` call is made to set TranslationService language. Since that call is synchronous, it is important to ensure that UserService has been initialized before fetching its data.
+--
++
+// TODO: Link to bootstrapRemoteComponent
+IMPORTANT: Providers passed in the `bootstrapRemoteComponent` method call should contain any providers required by the Remote Component. Any services or injection tokens have to be defined here. It is important that those providers are aligned with imports defined via the Remote Component's definition. Depending on the Remote Component, different providers and imports will be defined.
+
++
+Created application is going to have an Injector (just like a module does). This Injector will be used by the instance of ExampleComponent rendered using Web Components technology.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:495]: **The** created application...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 25]
```diff
@@ -0,0 +1,567 @@
...
++
+// TODO: Link to bootstrapRemoteComponent
+For Angular based Remote Components, it is recommended to use `bootstrapRemoteComponent` and define every required provider as an argument of this method. This approach will ensure that the rendered component has all required services, tokens, etc. already set up.
+// TODO: Correct link to exact resource for Angular issues
+// TODO: Link to bootstrapRemoteComponent
+Fixing Angular issues (Angular requirement only):::
+The `bootstrapRemoteComponent` method takes care of ngZone and platform sharing.
+Connecting router:::
+// TODO: Link to bootstrapRemoteComponent
+The `bootstrapRemoteComponent` method is responsible for connecting the Remote Component's router to Shell's router (if there is one defined), so their states are always the same. The connection is set up in the same way as for the <<module-rotuer-connection, module's router>>.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:505]: router to **the** Shell's router

File: docs/modules/general/pages/bootstrapping.adoc [hunk 26]
```diff
@@ -0,0 +1,567 @@
...
+Registering the Custom Element:::
+// TODO: Link to bootstrapRemoteComponent
+The `bootstrapRemoteComponent` method registers ExampleComponent in the Custom Elements Registry, so anytime `<example-remote-component>` is rendered, ExampleComponent should be instantiated.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:508]: ... method registers **the** ExampleComponent in ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 27]
```diff
@@ -0,0 +1,567 @@
...
+
++
+For technologies other than Angular, it is recommended to:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:511]: maybe use here a NOTE: ...

File: docs/modules/general/pages/bootstrapping.adoc [hunk 28]
```diff
@@ -0,0 +1,567 @@
...
+* AngularAuthModule for authorization mechanisms.
+* CommonModule for common Angular functionalities.
+* SharedModule with `PortalCoreModule.forMicroFrontend()` import for allowing OneCX components and services usage.
+* PortalCoreModule so the component recognizes OneCX components and services.
+* TranslateModule for translations mechanism.
+* AngularRemoteComponentsModule.
+--
++
+// TODO: Link to bootstrapRemoteComponent
+In the **example.component.bootstrap.ts**, some providers related to those dependencies were already declared in `bootstrapRemoteComponent` method call.
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:529]: ... were already declared in **the** `bootstrapRemoteComponent` method call.

File: docs/modules/general/pages/bootstrapping.adoc [hunk 29]
```diff
@@ -0,0 +1,567 @@
...
+.RemoteComponentConfig structure
+|===
+|**Property**|**Type**|**Description**
+|`appId` | `string` | Unique identifier of the Microfrontend Remote Component is part of.
+|`productName` | `string` | Name of the Application currently Remote Component is part of.
+|`permissions` | `string[]` | Current user permissions related to the Remote Component's Microfrontend.
+|`baseUrl` | `string` | URL of Remote Component's Microfrontend to be used when accessing its content (remote entry file, assets, etc.), e.g. `'/mfe/mfe_name'`.
+|===
++
+For technologies other than Angular, it is recommended to:
```
Comment (KimFFVII) [inline docs/modules/general/pages/bootstrapping.adoc:563]: maybe use here a NOTE: ...

File: docs/modules/general/pages/implementation-details/bootstrapping.adoc [hunk 30]
```diff
@@ -0,0 +1,584 @@
...
+. Webpack configuration needs to expose the *main.ts* file of the Remote Component.
+. The **main.ts** file of the Remote Component needs to bootstrap the component asynchronously.
+// TODO: Correct link to exact resource for Angular issues
+. Component bootstrap
+.. has to cover Angular issues (for Angular-based Remote Components only).
+.. needs to register the component as a Custom Element in the Custom Elements Registry.
+.. needs to adapt routing so Shell registers routing happening within the Remote Component.
+. Component class needs to use the Remote Component initialization mechanism.
+
+Below is a Remote Component setup example (Angular-based) for using the Webcomponent expose method. Each file's content is going to be explained in detail, with important context information for technologies other than those presented.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/bootstrapping.adoc:304]: Below is an example of setting up a remote component in Angular using the Webcomponent expose method.

File: docs/modules/general/pages/implementation-details/webpack.adoc [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+In the Angular example, there are two remote modules defined:
+
+* **ExampleModule** - Microfrontend's remote module pointing to Microfrontend's main.ts file.
+* **ExampleComponent** - Remote Component's remote module pointing to component's main.ts file.
+
+IMPORTANT: https://webpack.js.org/concepts/module-federation/[Module Federation] considers any exposed content as a **remote module**. In the OneCX context, Microfrontends expose two types of content, being modules and Remote Components.
+
+Shared resources::
+// TODO: Link to OneCX package sharing
+Each Microfrontend can define packages that can be shared with others. Every Microfrontend can specify what packages and with what versions are allowed to be shared. In the OneCX context, it is important to ensure the correct package sharing configuration dependent on used packages and functionality used within the Microfrontend's exposed content.
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/webpack.adoc:87]: proposition: it is important to ensure the correct package sharing configuration dependent on the used packages used functionalities within the Microfrontend's exposed content.

File: docs/modules/general/pages/implementation-details/bootstrapping.adoc [hunk 32]
```diff
@@ -0,0 +1,584 @@
+= OneCX Microfrontend Content Bootstrapping
+
+== Overview
+// TODO: Add link to MFE loading
```
Comment (anninowak) [inline docs/modules/general/pages/implementation-details/bootstrapping.adoc:4]: What do you think of a general introduction of Microfrontend Bootrapping?

Bootstrapping a microfrontend involves setting up the initial structure and configuration for a microfrontend application. This process typically includes creating the necessary directories, installing dependencies, and configuring the build tools to ensure that the microfrontend can run independently and integrate with other microfrontends or a shell application.


Would it be suitable in your opinion?

Comment (markuczy) [reply docs/modules/general/pages/implementation-details/bootstrapping.adoc:4]: I think its a perfect addition to the doc :)

**Pull request merged**